rust/pact_verifier/src/messages.rs

```diff
@@ -119,3 +121,29 @@ fn display_result(body_result: ANSIGenericString<str>, metadata_result: Vec<(Str
   }
   println!("      has a matching body ({})", body_result);
 }
+
+fn extract_metadata(actual_response: &Response) -> HashMap<String, String>{
+  let mut metadata = hashmap!{
+    "contentType".into() => actual_response.lookup_content_type().unwrap_or_default(),
+    "Content-Type".into() => actual_response.lookup_content_type().unwrap_or_default()
+  };
+
+  match actual_response.body.clone() {
+    OptionalBody::Present(ref body, _) => {
+      let m: MessageMetadata = serde_json::from_slice(body).unwrap_or(MessageMetadata{ metadata: hashmap!{} });
+      log::debug!("message metadata found in response, appending = {:?}", m);
```

Comment:
I think this is a `trace` level log.

Reply:
I think this is a `debug` level log.

Reply:
Following the guidelines we were talking about last week [recorded here](https://github.com/pact-foundation/pact-specification#logging-in-a-pact-library), I think this log is more for maintainers than it is for users (because the whole metadata structure is already logged elsewhere). 

If I'm reading it correctly - which I might not be - this is logging the construction of the metadata structure/object/hash (whatever it's called in rust), which is probably only useful when debugging pact, not when a user is debugging their tests.

Another test is the language in the log message -a user doesn't know what "appending" means in this context, but the maintainer(s) of this code do.

I think the best experience for users would have the metadata logged once, using a log message that requires no extra context.

Reply:
Let's leave this for another pass through. I did update another to trace though.

I think related to this, we should read through the info and debug logs for key scenarios and ask ourselves if it presents a coherent and clear view of what's happening. This way, we aren't just changing log levels, but are proactively giving the best experience for users to self diagnose, and we get good feedback if a bug request is raised.